packages/experimental/Expander/windows/ReactNativeExpander/ExpanderView.cpp

```diff
@@ -0,0 +1,272 @@
+#include "pch.h"
+#include "JSValueXaml.h"
+#include "ExpanderView.h"
+#include "ExpanderView.g.cpp"
+
+#include <limits>
+#include <stdexcept>
+
+namespace winrt {
+    using namespace Microsoft::ReactNative;
+    using namespace Windows::Foundation;
+    using namespace Windows::UI::Xaml::Media;
+}
+
+namespace winrt::ReactNativeExpander::implementation {
+
+    ExpanderView::ExpanderView(winrt::IReactContext const& reactContext) : m_reactContext(reactContext) {
+        auto expander = Microsoft::UI::Xaml::Controls::Expander();
+        this->VerticalContentAlignment(xaml::VerticalAlignment::Top);
+        this->Content(expander);
+
+        RegisterEvents();
+    }
+
+    void ExpanderView::RegisterEvents() {
+        auto expander = this->Content().try_as<Microsoft::UI::Xaml::Controls::Expander>();
+
+        m_expanderCollapsingRevoker = expander.Collapsed(winrt::auto_revoke,
+            [ref = get_weak()](auto const& sender, auto const& args) {
+            if (auto self = ref.get()) {
+                self->OnCollapsing(sender, args);
+            }
+        });
+        m_expanderExpandingRevoker = expander.Expanding(winrt::auto_revoke,
+            [ref = get_weak()](auto const& sender, auto const& args) {
+            if (auto self = ref.get()) {
+                self->OnExpanding(sender, args);
+            }
+        });
+    }
+
+    void ExpanderView::UpdateProperties(winrt::IJSValueReader const& reader) {
+        m_updating = true;
+        auto const& propertyMap = JSValueObject::ReadFrom(reader);
+        auto expander = this->Content().try_as<Microsoft::UI::Xaml::Controls::Expander>();
+
+        for (auto const& pair : propertyMap) {
+            auto const& propertyName = pair.first;
+            auto const& propertyValue = pair.second;
+
+            if (propertyName == "expandDirection") {
+                if (propertyValue.IsNull()) {
+                    expander.ClearValue(Microsoft::UI::Xaml::Controls::Expander::ExpandDirectionProperty());
+                }
+                else {
+                    auto direction = propertyValue.AsString();
+                    if (direction == "up") {
+                        expander.ExpandDirection(Microsoft::UI::Xaml::Controls::ExpandDirection::Up);
+                    }
+                    else {
+                        expander.ExpandDirection(Microsoft::UI::Xaml::Controls::ExpandDirection::Down);
+                    }
+                }
+            }
+            else if (propertyName == "expanded") {
+                if (propertyValue.IsNull()) {
+                    expander.ClearValue(Microsoft::UI::Xaml::Controls::Expander::IsExpandedProperty());
+                    m_expanded = false;
```

Comment:
`m_expanded` is only ever written to and therefore unnecessary, remove the member variable